src/libopensc/reader-pcsc.c

```diff
@@ -259,13 +259,13 @@ static int pcsc_internal_transmit(sc_reader_t *reader,
 		rv = priv->gpriv->SCardTransmit(card, &sSendPci, sendbuf, dwSendLength,
 				   &sRecvPci, recvbuf, &dwRecvLength);
 	} else {
-		if (priv->gpriv->SCardControlOLD != NULL) {
+		if (!priv->gpriv->SCardControlOLD) {
 			rv = priv->gpriv->SCardControlOLD(card, sendbuf, dwSendLength,
 				  recvbuf, &dwRecvLength);
 		}
-		else {
+		else if (!priv->gpriv->SCardControl) {
 			rv = priv->gpriv->SCardControl(card, (DWORD) control, sendbuf, dwSendLength,
```

Comment:
last nit from the clang is to merge these two lines to not have closing brace alone on the line and connect the cast with the variable name:
```suggestion
		} else if (!priv->gpriv->SCardControl) {
			rv = priv->gpriv->SCardControl(card, (DWORD)control, sendbuf, dwSendLength,
```
then we I think we are good